### DISCUSSION TITLE
Bug: `env` flags don't exist in flat config

### DISCUSSION THREAD

**Discussion by mlwyatt** [eslint/eslint#17101] [Q&A] [answered]
### Environment

Node version: 16.18.1
npm version: 8.19.2 (I use yarn instead)
  I use yarn version: 1.22.19
Local ESLint version: 8.38.0
Global ESLint version: N/A
Operating System:
  - Mac OS 13.0.1
  - Docker Desktop 4.18.0
  - Docker 20.10.24

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
export default [
  {
    rules: {
      'no-undef': ['error'],
      'no-console': ['error'],
    },
  },
];
```
</details>

```js
window.alert('Hi');
console.log(2);
Array.isArray(2);
Promise.resolve(2);
```

### What did you expect to happen?

```
  1:1  error  Unexpected console statement  no-console
```

### What actually happened?

```
  1:1  error  'window' is not defined       no-undef
  2:1  error  'console' is not defined      no-undef
  2:1  error  Unexpected console statement  no-console
```

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50ICduby11bmRlZic6IFsnZXJyb3InXSwgJ25vLWNvbnNvbGUnOiBbJ2Vycm9yJ10gKi9cbmNvbnNvbGUubG9nKDIpO1xuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJjb25zdHJ1Y3Rvci1zdXBlciI6WyJlcnJvciJdLCJmb3ItZGlyZWN0aW9uIjpbImVycm9yIl0sImdldHRlci1yZXR1cm4iOlsiZXJyb3IiXSwibm8tYXN5bmMtcHJvbWlzZS1leGVjdXRvciI6WyJlcnJvciJdLCJuby1jYXNlLWRlY2xhcmF0aW9ucyI6WyJlcnJvciJdLCJuby1jbGFzcy1hc3NpZ24iOlsiZXJyb3IiXSwibm8tY29tcGFyZS1uZWctemVybyI6WyJlcnJvciJdLCJuby1jb25kLWFzc2lnbiI6WyJlcnJvciJdLCJuby1jb25zdC1hc3NpZ24iOlsiZXJyb3IiXSwibm8tY29uc3RhbnQtY29uZGl0aW9uIjpbImVycm9yIl0sIm5vLWNvbnRyb2wtcmVnZXgiOlsiZXJyb3IiXSwibm8tZGVidWdnZXIiOlsiZXJyb3IiXSwibm8tZGVsZXRlLXZhciI6WyJlcnJvciJdLCJuby1kdXBlLWFyZ3MiOlsiZXJyb3IiXSwibm8tZHVwZS1jbGFzcy1tZW1iZXJzIjpbImVycm9yIl0sIm5vLWR1cGUtZWxzZS1pZiI6WyJlcnJvciJdLCJuby1kdXBlLWtleXMiOlsiZXJyb3IiXSwibm8tZHVwbGljYXRlLWNhc2UiOlsiZXJyb3IiXSwibm8tZW1wdHkiOlsiZXJyb3IiXSwibm8tZW1wdHktY2hhcmFjdGVyLWNsYXNzIjpbImVycm9yIl0sIm5vLWVtcHR5LXBhdHRlcm4iOlsiZXJyb3IiXSwibm8tZXgtYXNzaWduIjpbImVycm9yIl0sIm5vLWV4dHJhLWJvb2xlYW4tY2FzdCI6WyJlcnJvciJdLCJuby1leHRyYS1zZW1pIjpbImVycm9yIl0sIm5vLWZhbGx0aHJvdWdoIjpbImVycm9yIl0sIm5vLWZ1bmMtYXNzaWduIjpbImVycm9yIl0sIm5vLWdsb2JhbC1hc3NpZ24iOlsiZXJyb3IiXSwibm8taW1wb3J0LWFzc2lnbiI6WyJlcnJvciJdLCJuby1pbm5lci1kZWNsYXJhdGlvbnMiOlsiZXJyb3IiXSwibm8taW52YWxpZC1yZWdleHAiOlsiZXJyb3IiXSwibm8taXJyZWd1bGFyLXdoaXRlc3BhY2UiOlsiZXJyb3IiXSwibm8tbG9zcy1vZi1wcmVjaXNpb24iOlsiZXJyb3IiXSwibm8tbWlzbGVhZGluZy1jaGFyYWN0ZXItY2xhc3MiOlsiZXJyb3IiXSwibm8tbWl4ZWQtc3BhY2VzLWFuZC10YWJzIjpbImVycm9yIl0sIm5vLW5ldy1zeW1ib2wiOlsiZXJyb3IiXSwibm8tbm9ub2N0YWwtZGVjaW1hbC1lc2NhcGUiOlsiZXJyb3IiXSwibm8tb2JqLWNhbGxzIjpbImVycm9yIl0sIm5vLW9jdGFsIjpbImVycm9yIl0sIm5vLXByb3RvdHlwZS1idWlsdGlucyI6WyJlcnJvciJdLCJuby1yZWRlY2xhcmUiOlsiZXJyb3IiXSwibm8tcmVnZXgtc3BhY2VzIjpbImVycm9yIl0sIm5vLXNlbGYtYXNzaWduIjpbImVycm9yIl0sIm5vLXNldHRlci1yZXR1cm4iOlsiZXJyb3IiXSwibm8tc2hhZG93LXJlc3RyaWN0ZWQtbmFtZXMiOlsiZXJyb3IiXSwibm8tc3BhcnNlLWFycmF5cyI6WyJlcnJvciJdLCJuby10aGlzLWJlZm9yZS1zdXBlciI6WyJlcnJvciJdLCJuby11bmRlZiI6WyJlcnJvciJdLCJuby11bmV4cGVjdGVkLW11bHRpbGluZSI6WyJlcnJvciJdLCJuby11bnJlYWNoYWJsZSI6WyJlcnJvciJdLCJuby11bnNhZmUtZmluYWxseSI6WyJlcnJvciJdLCJuby11bnNhZmUtbmVnYXRpb24iOlsiZXJyb3IiXSwibm8tdW5zYWZlLW9wdGlvbmFsLWNoYWluaW5nIjpbImVycm9yIl0sIm5vLXVudXNlZC1sYWJlbHMiOlsiZXJyb3IiXSwibm8tdW51c2VkLXZhcnMiOlsiZXJyb3IiXSwibm8tdXNlbGVzcy1iYWNrcmVmZXJlbmNlIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtY2F0Y2giOlsiZXJyb3IiXSwibm8tdXNlbGVzcy1lc2NhcGUiOlsiZXJyb3IiXSwibm8td2l0aCI6WyJlcnJvciJdLCJyZXF1aXJlLXlpZWxkIjpbImVycm9yIl0sInVzZS1pc25hbiI6WyJlcnJvciJdLCJ2YWxpZC10eXBlb2YiOlsiZXJyb3IiXSwibm8tY29uc29sZSI6WyJlcnJvciJdfSwiZW52Ijp7ImVzNiI6dHJ1ZSwiYnJvd3NlciI6dHJ1ZX19fQ==

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

I know how to fix this using a `.eslintrc.js` file, but I'm trying to convert to a flat config file in preparation for the future.

I couldn't find anywhere in the [docs](https://eslint.org/docs/latest/use/configure/configuration-files-new) how to set those `env` flags to enable built-in globals like `console` and `window`.

Any help is appreciated. Thanks in advance!

<details>
<summary>.eslintrc.js Fix</summary>

```js
module.exports = {
  env: { broswer: true },
  rules: {
    'no-undef': ['error'],
    'no-console': ['error'],
  },
};
```
</details>

**Comment by aladdin-add** [comment, marked as answer]
you'll need to install and use the package `globals`, as described in https://eslint.org/blog/2022/08/new-config-system-part-2/#goodbye-environments%2C-hello-globals

**Reply by mlwyatt** [reply]
> you'll need to install and use the package `globals`, as described in https://eslint.org/blog/2022/08/new-config-system-part-2/#goodbye-environments%2C-hello-globals

This works just as expected.

However, shouldn't that either be documented somewhere on the [new config docs](https://eslint.org/docs/latest/use/configure/configuration-files-new) or linked to in the [globals section](https://eslint.org/docs/latest/use/configure/configuration-files-new#configuring-global-variables)

**Reply by fasttime** [reply]
> However, shouldn't that either be documented somewhere on the [new config docs](https://eslint.org/docs/latest/use/configure/configuration-files-new) or linked to in the [globals section](https://eslint.org/docs/latest/use/configure/configuration-files-new#configuring-global-variables)

@mlwyatt I'd love to see that documented in the [section about global variables](https://github.com/eslint/eslint/blob/6d8bffdf45d50e272dc45e6d2d05b4a737514468/docs/src/use/configure/configuration-files-new.md#configuring-global-variables) in the new config docs. If you want to take a shot at a PR and add what is missing, we would greatly appreciate it.
